pydantic_settings/sources.py

```diff
@@ -877,6 +892,49 @@ def _read_file(self, file_path: Path) -> dict[str, Any]:
             return yaml.safe_load(yaml_file)
 
 
+class AzureKeyVaultSettingsSource(PydanticBaseSettingsSource):
+    _secret_client: SecretClient  # type: ignore
+
+    def __init__(
+        self,
+        settings_cls: type[BaseSettings],
+        url: str,
+        credential: TokenCredential,  # type: ignore
+    ) -> None:
+        import_azure_key_vault()
+        super().__init__(settings_cls)
+        self._secret_client = SecretClient(vault_url=url, credential=credential)  # type: ignore
+
+    def get_field_value(self, field: FieldInfo, field_name: str) -> tuple[Any, str, bool]:
+        field_value: Any | None = None
+
+        # It's not possible to use underscores in Azure Key Vault
+        secret_name = field_name.replace('_', '-')
+
+        try:
+            secret = self._secret_client.get_secret(secret_name)
+            field_value = secret.value
+        except ResourceNotFoundError:  # type: ignore
+            field_value = None
+
+        return field_value, field_name, False
+
+    def prepare_field_value(self, field_name: str, field: FieldInfo, value: Any, value_is_complex: bool) -> Any:
```

Comment:
Do you need to return the fetched value? what about the complex values(like json strings or nested models values).

Probably you don't need to override this method and use the parent one

Reply:
Overrided method removed.